avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Invoke-ResourceRemoval.ps1

```diff
@@ -185,6 +185,43 @@ function Invoke-ResourceRemoval {
             }
             break
         }
+        'Microsoft.Subscription/aliases' {
+            #Update-Module -Name Az.Accounts -Force
+            #Update-Module -Name Az.Subscription -Force
+            #Import-Module Az.Subscription -Force
+
+            $subscriptionName = $ResourceId.Split('/')[4]
+            $subscription = Get-AzSubscription | Where-Object {$_.Name -eq $subscriptionName}
+            $subscriptionId =  $subscription.Id
+            $subscriptionState = $subscription.State
+
+            $null = Select-AzSubscription -SubscriptionId $subscriptionId -WarningAction SilentlyContinue
+
+            if ($PSCmdlet.ShouldProcess("Subscription [$subscriptionName]", 'Remove')) {
+                # Unregister resource providers and features
+                Write-Verbose "Cleanup registered resource providers and features..."
+                $null = Unregister-AzProviderFeature -FeatureName "ArcServerPrivateLinkPreview" -ProviderNamespace "Microsoft.HybridCompute"
+                $null = Unregister-AzProviderFeature -FeatureName "AzureServicesVm" -ProviderNamespace "Microsoft.AVS"
+                $null = Unregister-AzResourceProvider -ProviderNamespace "Microsoft.HybridCompute"
+                $null = Unregister-AzResourceProvider -ProviderNamespace "Microsoft.AVS"
```

Comment:
Does this target a specific subscription via the `Select-Azsubscription` command above? If so, it's fine - if it would instead tackle the environment itself it may break something that was previously registered for other reasons. Just want to make sure

Reply:
One thing I'd raise though is that is is very 'hard-coded' & static. If you need to add anything in the future, you need to return also to the removal script & update this. This in turn would re-run all workflows of all modules, etc. etc.

I wonder if it makes sense to just run a generic unregistration of all ProviderFeatures & ResourceProviders for the subscription you created (if as per the first comment, that's the one that is targeted)

Reply:
Hey @sebassem, I can see you removed the 'un'-registration entirely. Does it not need to be removed afterall_

Reply:
Answered offline. The jist of it - as the subscription is recreated each time, the unregistration isn't required